lib/motion/revision_calculator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "digest"
+require "motion"
+
+module Motion
+  class RevisionCalculator
+    attr_reader :revision_paths
+
+    def initialize(revision_paths:)
+      @revision_paths = revision_paths
+    end
+
+    def perform
+      derive_md5_hash
+    end
+
+    private
+
+    def file_contents
+      files.map { |f| File.read(f) }.join
+    end
+
+    def existent_paths
+      @existent_paths ||= revision_paths.all_paths.flat_map(&:existent)
+    end
+
+    def files
+      @files ||= existent_paths.flat_map { |path| Dir["#{path}/**/*", path].reject { |f| File.directory?(f) } }.uniq
+    end
+
+    def derive_md5_hash
+      Digest::MD5.hexdigest(file_contents)
```

Comment:
I don't think it technically matters for our purposes, but MD5 is considered a weak hash these days. Can we use `SHA256` or `SHA512` instead?

Reply:
In my research, I found that MD5 is faster than SHA256 and is often used for checksums on files and things that don't require security. With that in mind, do you still want me to use SHA256 or SHA512?

Reply:
I'm convinced! Let's stick with MD5. There is no security implication here.